conference/accounts.py

```diff
@@ -51,12 +68,253 @@ def clean(self):
         return data
 
 
+def signup_step_1_create_account(request) -> [TemplateResponse, redirect]:
+    """
+    Creates new account in the system, populating both auth.User and
+    assopy.AssopyUser
+    """
+
+    if request.user.is_authenticated:
+        return redirect('user_panel:dashboard')
+
+    form = NewAccountForm()
+
+    if request.method == 'POST':
+        form = NewAccountForm(data=request.POST)
+
+        if form.is_valid():
+            data = form.cleaned_data
+            with transaction.atomic():
+                assopy_user = AssopyUser.objects.create_user(
+                    email=data['email'],
+
+                    # Python conferences are on the first name basis, and since
+                    # we want to populate only one field (we don't want to
+                    # replace the builtin User model yet) we populate the first
+                    # name only.
+                    first_name=data['name'],
+
+                    password=data['password1'],
+                )
+                current_site = get_current_site(request)
+                send_verification_email(assopy_user, current_site)
+
+                messages.success(request, "Email verification sent")
+
+            return redirect('accounts:signup_step_2_please_verify_email')
+
+    return TemplateResponse(request, "ep19/bs/accounts/signup.html", {
+        'form': form,
+        'next': request.GET.get('next', '/'),
+    })
+
+
+def signup_step_2_please_verify_email(request):
+    return TemplateResponse(
+        request, "ep19/bs/accounts/signup_please_verify_email.html", {}
+    )
+
+
+def send_verification_email(assopy_user, current_site) -> None:
+
+    new_token = create_new_email_verification_token(assopy_user.user)
```

Comment:
the way I think about it in this context, is that token is 'accounts' specific, rather than 'model' specific, so I put the generation closer to the actual code that uses it.